src/unix/linux-core.c

```diff
@@ -1015,14 +1025,226 @@ uint64_t uv_get_total_memory(void) {
 }
 
 
-static uint64_t uv__read_cgroups_uint64(const char* cgroup, const char* param) {
-  char filename[256];
+/*
+ * Holds information about a given cgroups subsystem.
+ */
+typedef struct {
+  /* cgroups version, one of CGROUPS_VERSION_*. */
+  uint8_t cgroups_version;
+  /* Path in which param files are found for a subsystem. */
+  char* path;
+} uv__cgroups_subsystem_info_t;
+
+
+/*
+ * Given a `separator`-delimited string `haystack`, return 1 if `needle` exactly
+ * matches at least one of the elements in that sequence.
+ */
+static int uv__find_in_delimited_string(const char* haystack, const char* needle, const char* separator) {
+  char* haystack_mutable;
+  size_t haystack_size;
+  char* candidate;
+  char* haystack_ptr;
+  haystack_size = strlen(haystack) + 1;
+  if (needle == NULL || strlen(needle) >= haystack_size)
+    return 0;
+  haystack_mutable = malloc(haystack_size);
+  uv__strscpy(haystack_mutable, haystack, haystack_size);
+  haystack_ptr = haystack_mutable;
+  do {
+    candidate = strsep(&haystack_ptr, separator);
+    if (strcmp(candidate, needle) == 0) {
+      free(haystack_mutable);
+      return 1;
+    }
+  } while (haystack_ptr != NULL);
+  free(haystack_mutable);
+  return 0;
+}
+
+
+/*
+ * Read /proc/self/mountinfo and /proc/self/cgroup to get info about how the
```

Comment:
putthing some sample output of those 2 files here as documentation would be helpful